commcare-cloud/commcare_cloud/commands/command_base.py

```diff
@@ -0,0 +1,20 @@
+import abc
+import six
+
+# this is just to document intent; the subclass will use @classmethod
+abstractclassmethod = abc.abstractmethod
+
+
+@six.add_metaclass(abc.ABCMeta)
+class CommandBase(object):
+    command = None
+    help = None
+    aliases = ()
+
+    @abstractclassmethod
+    def make_parser(cls, parser):
```

Comment:
I think it would be cleaner if these class methods were normal instance methods. I used `classmethod` in [`Lookup` & friends](https://github.com/dimagi/commcare-cloud/blob/dd63996ffe6d58194a25cc42ce650f004c3acb66/commcare-cloud/commcare_cloud/commands/inventory_lookup/inventory_lookup.py#L14) because they needed a reference to the parser, but in hindsight it feels kinda dirty to do it that way because it modifies global state (the class object). I think it would be cleaner if [`commcare_cloud.main`](https://github.com/dimagi/commcare-cloud/blob/dd63996ffe6d58194a25cc42ce650f004c3acb66/commcare-cloud/commcare_cloud/commcare_cloud.py#L42-L47) were written like this:
```py
    commands = {}
    for StandardArg in STANDARD_ARGS:
        # note: make_parser became or is called by __init__ here
        cmd = StandardArg(subparsers.add_parser(
            cmd.command,
            help=cmd.help,
            aliases=cmd.aliases,
        ))
        commands[cmd.command] = cmd
        for alias in cmd.aliases:
            commands[alias] = cmd

    args, unknown_args = parser.parse_known_args()
    commands[args.command].run(args, unknown_args)
```